File: GenHash.java
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;

public class GenHash {
    public static void main(String[] args) {
        BCryptPasswordEncoder encoder = new BCryptPasswordEncoder();
        String hash = encoder.encode("password123");
        System.out.println("Hash: " + hash);
        System.out.println("Verify: " + encoder.matches("password123", hash));
    }
}

File: gen_hash.php
<?php
// Generate BCrypt hash for password123 using PHP
$password = 'password123';
$hash = password_hash($password, PASSWORD_BCRYPT, ['cost' => 10]);
echo "Hash for 'password123': " . $hash . "\n";

// Verify it works
$ok = password_verify($password, $hash);
echo "Verification: " . ($ok ? "OK" : "FAILED") . "\n";
